File: src/main/java/com/serdeliverance/cryptowallet/dto/UserDTO.java
package com.serdeliverance.cryptowallet.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Optional;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class UserDTO {
    private Optional<Integer> id;
    private String username;
    private String email;
}

File: src/main/kotlin/com/serdeliverance/cryptowallet/adapter/in/web/UserDTO.kt
package com.serdeliverance.cryptowallet.adapter.`in`.web

import com.serdeliverance.cryptowallet.v2.domain.User
import java.util.*

data class UserDTO(val id: Int?, val username: String, val email: String) {

    companion object {
        fun User.toDTO() = UserDTO(
            id = this.id,
            username = this.username,
            email = this.email
        )
    }
}
